### DISCUSSION TITLE
Setting global, multiple logging levels

### DISCUSSION THREAD

**Discussion by cubukinas** [rabbitmq/chef-cookbook#599] [Q&A] [answered]
I am trying to avoid logs such as:
`/var/log/rabbit@rabbitetc.log`
 ```
  2022-11-09 18:28:03.779 [info] <0.xxxxxx.x> accepting AMQP connection <0.XXXXX.X> (xx.xxx.xxx.xx:xxxxx -> xx.xxx.xx.xx:xxxx)
2022-11-09 18:28:03.781 [info] <0.xxxxxx.2> closing AMQP connection <0.25212.2> (xx.xxx.xxx.xx:xxxxx -> xx.xxxx.xxx.xx:xxxx, vhost: 'xxxx', user: 'xxxx')
2022-11-09 18:28:03.781 [info] <0.xxxxx.2> Closing all channels from connection 'xx.xxx.xxx.xxx:xxxxx -> xxx.xxx.xx.xxx:xxxx' because it has been closed
 ```

Tried setting
`node.default['rabbitmq']['log_levels'] = { 'connection' => 'warning' }`

The config was changed, but we are still receiving these logs after rabbitmq-server restart
` {log_levels, [{ connection, warning }]},`

Is it possible to configure global configuration via this cookbook, or set multi level global and per-category, e.g:
```
            {log_levels, [{connection, warning}, {queue, error}]},
            {log,
                [{file, [{level, warning}]},
                {categories,
                    [{connection,
                        [{level, warning}]
                    },
                    {queue,
                        [{level, error}]
                    }
                    ]
                }]
            },
```
Any help would be appreciated, thanks in advance.

**Comment by michaelklishin** [comment, marked as answer]
You can provide [arbitrary configuration in the Erlang term format](https://github.com/rabbitmq/chef-cookbook/blob/v5.x/templates/default/rabbitmq.config.erb#L149).

However, modern RabbitMQ versions use a different logging library and [most examples have been moved to the `rabbitmq.conf` format](https://www.rabbitmq.com/logging.html). So you have to set [`rabbit.log.file.level`](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/priv/schema/rabbit.schema#L1787) as a nested Erlang term, what this cookbook generates targets RabbitMQ before `rabbitmq.conf` and before the logging library it uses today.

**Comment by michaelklishin** [comment]
#497 and #558 are highly relevant for this discussion.
